activiti-cloud-services-org/activiti-cloud-services-org-service/src/main/java/org/activiti/cloud/services/organization/service/ModelService.java

```diff
@@ -210,23 +228,108 @@ public Model updateModelContent(Model modelToBeUpdate,
                 .flatMap(validator -> validator.convertToModelContent(fixedFileContent.getFileContent()))
                 .ifPresent(modelContent -> modelToBeUpdate.setTemplate(modelContent.getTemplate()));
 
-        return modelRepository.updateModelContent(modelToBeUpdate,
-                                                  fixedFileContent);
+        return modelRepository.updateModelContent(modelToBeUpdate, fixedFileContent);
     }
 
-    public FileContent checkAndFixModelContent(Model model,
-                                               FileContent fileContent) {
-        return modelContentService
-                .findModelContentConverter(model.getType())
-                .map(converter -> converter
-                        .convertAndFixModelContentId(fileContent.getFileContent(),
-                                                     modelContentService.getModelContentId(model)))
-                .map(content -> new FileContent(fileContent.getFilename(),
-                                                fileContent.getContentType(),
-                                                content))
-                .orElse(fileContent);
+
+  public FileContent checkAndFixModelContent(Model model, FileContent fileContent) {
+      FileContent fixedFileContent = null;
+      switch (model.getType()){
+        case "PROCESS": {
+          fixedFileContent = this.createFixedProcessFileContent(model, fileContent);
+          break;
+        }
+        case "FORM":{
+          fixedFileContent = this.createFixedFormModelFileContent(model, fileContent);
+          break;
+        }
+        default:
+          fixedFileContent = this.createFixedUIModelFileContent(model, fileContent);
+      }
+      return fixedFileContent;
+  }
+
+  private FileContent createFixedProcessFileContent(Model model, FileContent fileContent) {
```

Comment:
instead of "Fixed" can we put something more explicit such as: "..OverridingId" "..SettingNewId" etc ?